ir/instr.cpp

```diff
@@ -700,52 +721,74 @@ void TernaryOp::print(ostream &os) const {
   case FShr:
     str = "fshr ";
     break;
+  case FMA:
+    str = "fma ";
+    break;
   }
 
-  os << getName() << " = " << str << *a << ", " << *b << ", " << *c;
+  os << getName() << " = " << str << fmath << *a << ", " << *b << ", " << *c;
 }
 
 StateValue TernaryOp::toSMT(State &s) const {
   auto &av = s[*a];
   auto &bv = s[*b];
   auto &cv = s[*c];
-  function<expr(const expr&, const expr&, const expr&)> fn;
+  function<StateValue(const expr&, const expr&, const expr&)> fn;
 
   switch (op) {
   case FShl:
-    fn = [](auto a, auto b, auto c) {
-      return expr::fshl(a, b, c);
+    fn = [&](auto a, auto b, auto c) -> StateValue {
+      return { expr::fshl(a, b, c), true };
     };
     break;
+
   case FShr:
-    fn = [](auto a, auto b, auto c) {
-      return expr::fshr(a, b, c);
+    fn = [&](auto a, auto b, auto c) -> StateValue {
+      return { expr::fshr(a, b, c), true };
+    };
+    break;
+
+  case FMA:
+    fn = [&](auto a, auto b, auto c) -> StateValue{
+      return fm_poison(s, a, b, c, [](expr &a, expr &b, expr &c) { 
+                                   return expr::fma(a, b, c); }, fmath, false);
     };
     break;
   }
 
   if (getType().isVectorType()) {
-    vector<StateValue> vals;
```

Comment:
this change seems unrelated

Reply:
Changed back to how it was before. This was only a line swap for it to be consistent with BinOp::toSMT().